src/cli/commands/add.js

```diff
@@ -34,11 +34,13 @@ export class Add extends Install {
     ]
       .filter(Boolean)
       .shift();
+    this.ignoreDependencyTypeWarning = config.commandName === 'upgrade' || config.commandName === 'upgrade-interactive';
```

Comment:
nit: I'd move this into a constant array at the beginning of the file (like `SILENCE_DEPENDENCY_TYPE_WARNINGS`), and instead of storing the computation in the class instance I'd just put a `array.indexOf() === -1` in the if condition. What do you think?

Reply:
Done 👍